src/gui/graphdesignerpage.cpp

```diff
@@ -71,12 +64,32 @@ void GraphDesignerPage::slotEdgeAttrs() {
     new GraphAttrsDlg(this, AttrsType::Edges);
 }
 
+void GraphDesignerPage::slotGraphGen() {
+    if (m_numNodes > 0) {
+        QMessageBox msgBox;
+        msgBox.setText("You are attempting to create a new graph, this will reset your current progress.");
+        msgBox.setInformativeText("Do you want to continue?");
+        msgBox.setStandardButtons(QMessageBox::Ok | QMessageBox::Cancel);
+        msgBox.setDefaultButton(QMessageBox::Ok);
+        if (msgBox.exec() == QMessageBox::Ok) {
+            graphGen();
+        }
+    }
+    else if (!this->m_nodeAttrScope.isEmpty()) {
+        graphGen();
+    }
+    else {
+        QMessageBox::warning(this, "Graph Generator",
+            "Make sure you have set valid node attributes before attempting to open the graph generator");
+    }
+}
+    
 void GraphDesignerPage::slotNodeAttrs()
 {
     new GraphAttrsDlg(this, AttrsType::Nodes);
 }
 
-void GraphDesignerPage::slotGraphGen()
+void GraphDesignerPage::graphGen()
 {
     new GraphGenDlg(this, m_mainGUI);
 }
```

Comment:
why do we need this function? might be better to call the `new GraphGenDlg` straight away when needed